src/_pytest/fixtures.py

```diff
@@ -525,9 +526,10 @@ def fspath(self) -> py.path.local:
         return self._pyfuncitem.fspath  # type: ignore
 
     @property
-    def keywords(self):
+    def keywords(self) -> MutableMapping[str, Any]:
         """Keywords/markers dictionary for the underlying node."""
-        return self.node.keywords
+        node: nodes.Node = self.node
+        return node.keywords
```

Comment:
I think adding `def node(self) -> nodes.Node:` above would satisfy typecheckers, and I'd prefer it to modifying this function body.

Reply:
Right, I was tempted to add it but didn't because

1. `nodes.Node` itself is not a type we intend to make public, but could use `Union[Item, Collector]` instead

2. Even `Union[Item, Collector]` would not be great - often the user does `request.node` in a known context and would assume the `node` is a specific node type (like `Function`) and then access it that way, but if we make the type `Union[Item, Collector]` it would suddenly generate a type error and require an assert. So for now I just keep it `Any`.